### ISSUE TITLE
Support value numbering of SIMD types in RyuJIT optimizer

### ISSUE THREAD

**Issue by cmckinsey** [dotnet/runtime#5859]
Per the comment, extend value numbering to support SIMD types in RyuJIT.

valuenum.cpp

```

#ifdef FEATURE_SIMD
    // TODO-CQ: For now TYP_SIMD values are not handled by value numbering to be amenable for CSE'ing.
    if (oper == GT_SIMD) 
    {
        tree->gtVNPair.SetBoth(vnStore->VNForExpr(TYP_UNKNOWN));
        return;
    }
#endif


```

category:cq
theme:vector-codegen
skill-level:intermediate
cost:medium

**Comment by turbedi** [comment]
This issue can be closed since #31834 has been merged, right?

**Comment by tannergooding** [comment]
Yep, we've had VN support for a bit now.
